packages/graph/hash_graph/postgres_migrations/V4__ontology_functions.sql

```diff
@@ -45,60 +31,54 @@ OR REPLACE FUNCTION update_ontology_id (
   "updated_by_id" UUID
 ) RETURNS TABLE (version_id UUID, owned_by_id UUID) AS $update_ontology_id$
 DECLARE
-  _version_id UUID;
+  "_version_id" UUID;
 BEGIN
-  INSERT INTO version_ids (
-    version_id
-  ) VALUES (
-    gen_random_uuid()
-  ) RETURNING version_ids.version_id INTO _version_id;
-
-  SET CONSTRAINTS type_ids_pkey DEFERRED;
-  SET CONSTRAINTS type_id_unique DEFERRED;
   SET CONSTRAINTS type_ids_overlapping DEFERRED;
 
+  _version_id := gen_random_uuid();
+
   RETURN QUERY
   UPDATE type_ids
   SET
-    "transaction_time" = tstzrange(now(), NULL, '[)'),
-    "version" = update_ontology_id.version,
     "version_id" = _version_id,
-    "updated_by_id" = update_ontology_id.updated_by_id
+    "version" = update_ontology_id.version,
+    "updated_by_id" = update_ontology_id.updated_by_id,
+    "transaction_time" = tstzrange(now(), NULL, '[)')
   WHERE type_ids.base_uri = update_ontology_id.base_uri
     AND type_ids.transaction_time @> now()
-  RETURNING type_ids.version_id, type_ids.owned_by_id;
+  RETURNING _version_id, type_ids.owned_by_id;
 
-  SET CONSTRAINTS type_ids_pkey IMMEDIATE;
-  SET CONSTRAINTS type_id_unique IMMEDIATE;
   SET CONSTRAINTS type_ids_overlapping IMMEDIATE;
 END $update_ontology_id$ VOLATILE LANGUAGE plpgsql;
 
 CREATE
-OR REPLACE FUNCTION "update_type_ids_trigger" () RETURNS TRIGGER AS $pga$
-    BEGIN
-      IF (OLD.version != NEW.version - 1) THEN
-        RAISE EXCEPTION 'Not updating the latest id: % -> %', OLD.version, NEW.version
-        USING ERRCODE = 'invalid_parameter_value';
-      END IF;
+OR REPLACE FUNCTION "update_type_ids_trigger" () RETURNS TRIGGER AS $update_type_ids_trigger$
+BEGIN
+  IF (OLD.version != NEW.version - 1) THEN
+    RAISE EXCEPTION 'Not updating the latest id: % -> %', OLD.version, NEW.version
+    USING ERRCODE = 'invalid_parameter_value';
+  END IF;
+
+  INSERT INTO type_ids (
+    "version_id",
+    "base_uri",
+    "version",
+    "owned_by_id",
+    "updated_by_id",
+    "transaction_time"
+  ) VALUES (
+    NEW.version_id,
+    NEW.base_uri,
+    NEW.version,
+    NEW.owned_by_id,
+    NEW.updated_by_id,
+    NEW.transaction_time
+  );
 
-      INSERT INTO type_ids (
-        "base_uri",
-        "version",
-        "version_id",
-        "owned_by_id",
-        "updated_by_id",
-        "transaction_time"
-      ) VALUES (
-        OLD.base_uri,
-        OLD.version,
-        OLD.version_id,
-        OLD.owned_by_id,
-        OLD.updated_by_id,
-        tstzrange(lower(OLD.transaction_time), lower(NEW.transaction_time), '[)')
-      );
+  OLD.transaction_time = tstzrange(lower(OLD.transaction_time), lower(NEW.transaction_time), '[)');
 
-      RETURN NEW;
-    END$pga$ VOLATILE LANGUAGE plpgsql;
+  RETURN OLD;
```

Comment:
What is the implication of returning `OLD` here? I see that `update_entity_version_trigger` for knowledge returns `NEW`. Do we need it somewhere in code?

Reply:
The returned record of a `BEFORE UPDATE` trigger is the value the row will be updated to. I reverted the logic here to avoid the usage of `DEFERRED` constraints.